utils/src/abi_encoder/utils/math.ts

```diff
@@ -19,10 +19,7 @@ function sanityCheckBigNumberRange(
     }
 }
 function bigNumberToPaddedBuffer(value: BigNumber): Buffer {
-    const valueHex = `0x${value.toString(constants.HEX_BASE)}`;
-    const valueBuf = ethUtil.toBuffer(valueHex);
-    const valueBufPadded = ethUtil.setLengthLeft(valueBuf, constants.EVM_WORD_WIDTH_IN_BYTES);
-    return valueBufPadded;
+    return ethUtil.setLengthLeft(ethUtil.toBuffer(new ethUtil.BN(value.toFixed(0))), constants.EVM_WORD_WIDTH_IN_BYTES);
```

Comment:
It's like 13x faster on my machine but it doesn't seem like this is the bottleneck in encoding.